datashader/core.py

```diff
@@ -1227,9 +1227,14 @@ def validate_ranges(self, x_range, y_range):
         self.x_axis.validate(x_range)
         self.y_axis.validate(y_range)
 
+    def validate_extend(self, width, height):
+        if width == 0 or height == 0:
```

Comment:
```suggestion
        if width <= 0 or height <= 0:
```
Just in case someone passes in a negative number.